doc/development/branch-model.rst

```diff
@@ -12,7 +12,7 @@
   source code of HEAD always reflects a *production-ready* state. Each commit
   to master is considered a new release and will be tagged.
 
-The |PyInstaller| project doesn't use long living branches (beside `master`
+The PyInstaller project doesn't use long living branches (beside `master`
```

Comment:
This is not entirely true due to the `v4` branch... and `master` branch is actually unused.

Reply:
True. I'm going to leave this for now though; this PR only represents the changes needed for v5, not the entirety of what I'd like to do with the docs. Got plenty more rewriting to do.